SirinVoiting.sol

```diff
@@ -0,0 +1,162 @@
+
+    /**
+     * @dev Moves `amount` tokens from `sender` to `recipient` using the
+     * allowance mechanism. `amount` is then deducted from the caller's
+     * allowance.
+     *
+     * Returns a boolean value indicating whether the operation succeeded.
+     *
+     * Emits a `Transfer` event.
+     */
+    function transferFrom(address sender, address recipient, uint256 amount) external returns (bool);
+
+    /**
+     * @dev Emitted when `value` tokens are moved from one account (`from`) to
+     * another (`to`).
+     *
+     * Note that `value` may be zero.
+     */
+    event Transfer(address indexed from, address indexed to, uint256 value);
+
+    /**
+     * @dev Emitted when the allowance of a `spender` for an `owner` is set by
+     * a call to `approve`. `value` is the new allowance.
+     */
+    event Approval(address indexed owner, address indexed spender, uint256 value);
+}
+
+contract SirinVoting {
+
+     struct SRNVoter {
+        bool voted;
+        bool vote;
+        address voterAddress;
+    }
+
+    //////////////////////////////////////////////////////////
+    // Members
+    //////////////////////////////////////////////////////////
+
+    mapping(address => SRNVoter) public voters;
+    string public question = "is gilad gay?";
+    uint8 public totalVoters;
+
+    IERC20 public token;
+    uint256 public startTime;
+    uint256 public endTime;
+
+    //////////////////////////////////////////////////////////
+    // Event
+    //////////////////////////////////////////////////////////
+
+    event votedYesEvent(address voter);
+    event votedNoEvent(address voter);
+
+    //////////////////////////////////////////////////////////
+    // constructor
+    //////////////////////////////////////////////////////////
+
+    constructor(uint256 _startTime, uint256 _endTime, address _token) public {
+        //require(_startTime >= now);
+        require(_endTime >= _startTime);
+        require(_token != address(0));
+
+        startTime = _startTime;
+        endTime = _endTime;
+        token = IERC20(_token);
+    }
+
+    //////////////////////////////////////////////////////////
+    // public functions
+    //////////////////////////////////////////////////////////
+
+    function voteYes() public {
+        vote(true);
+    }
+
+    function voteNo() public {
+        vote(false);
+    }
+
+    // @return true if voting event has ended
+    function hasEnded() public view returns (bool) {
+        return now > endTime;
+    }
+
+    // @return true if voting event has started
+    function hasStarted() public view returns (bool) {
+        return now > startTime;
+    }
+
+    //////////////////////////////////////////////////////////
+    // private functions
+    //////////////////////////////////////////////////////////
+
+    function vote(bool voteResult) private {
+
+        SRNVoter storage sender = voters[msg.sender]; // assigns reference
+        require(!sender.voted);
```

Comment:
the requirement is to allow the same address to vote more than once but only count the last vote.